### ISSUE TITLE
CMake building for bare metal

### ISSUE THREAD

**Issue by niansa** [ivmai/bdwgc#393]
Hey, which cmake options do I need to pass to build for bare metal?

**Comment by ivmai** [comment]
I've not heard someone tried to build for bare metal using CMake. Is there any issue if no specific option is passed?

**Comment by ivmai** [comment]
Are you using Musl?

**Comment by niansa** [comment]
Nope, just some super-minimal C and C++ standard library without any posix stuff. Just malloc and free for memory management as well as some mmap/munmap-like functions.

**Comment by ivmai** [comment]
It would be interesting to know the results of your experiments. Patches are welcomed too!

**Comment by ivmai** [comment]
> Just malloc and free for memory management as well as some mmap/munmap-like functions.

The minimum you need is malloc (i.e. a way to get memory from the underlying subsystem).

A porting guide might be helpful for you - [porting.md](https://github.com/ivmai/bdwgc/blob/master/doc/porting.md)
